invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -102,8 +104,9 @@ def register(self, record, pid, **kwargs):
         try:
             pid.register()
             # Set metadata for DOI
+            doc = RDMDataCite43JSONSerializer.serialize_object(record)
             self.client.public_doi(
-                metadata=record, url="PIDS-FIXME.com", doi=pid.pid_value)
+                metadata=doc, url="PIDS-FIXME.com", doi=pid.pid_value)
```

Comment:
We might need a new endpoint: ``https://127.0.0.1:5000/doi/<doi>`` for the URL, that all it does is redirect. I got another use case today, about what if a DOI is assigned and then the record is changed to embargoed.

Reply:
Opened an issue for it, will treat it in another task/PR:
- Endpoint: https://github.com/inveniosoftware/invenio-rdm-records/issues/540
- Embargoed: https://github.com/inveniosoftware/invenio-rdm-records/issues/541 On this more comments on the above answer :)